### PR TITLE
All species (Except synthetics and plasmamen) can now be changelings

### LINKED ISSUE #5662: Vox changelings
**Problem Description**:
A vox did roll changeling, and was able to DNA sting people into becoming vox.

**What did you expect to happen**: 
Vox changelings not to convert people into voxes

**What happened instead**:
We did get people converted to vox, and dying from the lack of voxygen

**Why is this bad/What are the consequences:**
Vox are normally not transformable into, they are also behind a karma lock, and people dying since they don't have access to voxygen.

**Steps to reproduce the problem**:
Have a changeling start as vox, as they can't body swap or absorb them, have them DNA sting someone.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: The DNA also might have been spread by the changelings channel ability, letting other changelings convert people to voxes.

### PR DESCRIPTION
**Pull request by LittleBigKid2000**
All species can now be changelings, and all non-synthetic species can be absorbed because bluespace.
Fixes #5662?
Parts of this stolen from #5704
:cl:
tweak: All species except IPCs and plasmamen can now be changelings and affected by their genetic abilities.
/:cl:

### CODE REVIEW

**Comment by MarsM0nd** [comment]
Should probably also stop Dionas, as they are also a species that's unable to get changed/bodyswapped into as far as i know.

**Comment by Fox-McCloud** [comment]
This should probably be a check on DNA sting, to be honest.

**Comment by Rapidvalj** [comment]
You can absolutely absorb vox. It makes zero sense to make them not able to be changelings.

**Comment by tristan1333** [comment]
Don't merge this. It's shit. I'd rather everyone be able to be absorbed than outright removing the possibility of races playing.

**Comment by LittleBigKid2000** [comment]
I just tested it a second ago. DNA sting does not work on Vox. Absorbing does not work on Vox. Swap forms does not work on Vox.

**Comment by fludd12** [comment]
@tristan1333 IPCs already can't be changelings for similar reasons. This makes sense, at the least.

**Comment by IcyV** [comment]
What "makes sense" should always come second to mechanics and potential fun. Restricting entire species from a gamemode is punishing people for no real great reason.

Actually fix the core issue instead of bandaging the problem by taking things out.

**Comment by uraniummeltdown** [comment]
Hell no. I was the Vox ling Kikrataahkskrehi turning people into Voxes in that recent round, and it was me using the Transformation Sting in an interesting way. What should happen is make Vox and Dionae able to be DNA Stinged and absorbed, restricting races from antags is always boring. Plus it was hilarious seeing the crew dashing straight to Medbay for nitrogen.

**Comment by Fullofskittles** [comment]
I also think having races that are immune to lings is the opposite of what you want. Every race should be able to be absorbed, no one should get a free pass from something that is supposed to be scary and able to become literally anyone.

I would much rather the underlying issue be fixed instead of slapping a tiny bandage on it and calling it good. If anything turning crew into vox with transformation sting sounds like a clever way to accomplish your goal. :package: 

:-1:

**Comment by Doukhan** [comment]
Most people didn't have a problem with IPCs being unvamp-able because it made sense with them being machines and all.
Vox and diona though? They're (probably) flesh and blood, or sap and bark or something.

They're still organic and it's an unnecessary restriction IMO, just make them absorbable so there aren't more antag proof species running around.

**Comment by KingPhilipIII** [comment]
I don't know why but this reminds me of the time we had a vampire AI. Apparently it was a bug and the admins just went with it. But yea. I don't think they should be immune to being absorbed if they are able to become changelings, and making two more races barred from the vicious maintenance-tunnel-dwelling-face-eating-monsters isn't much of an improvement either.

**Comment by Fox-McCloud** [comment]
This is going to break things.

**Comment by LittleBigKid2000** [comment]
I had a feeling it would. I'll see if I can make it not break things tomorrow. Or some time in the future.

**Comment by uraniummeltdown** [comment]
Just tested Transforming and Transformation Sting with different species as a ling, if a ling gets a hold of weird DNA (Vox, Plasmaman, Slime Person, Diona) either by spawning as them or Hive Channel DNA, they can seamlessly Transform between species, they don't suddenly die because of issues with blood or organs or whatever. Transformation stings applies to other people also turns them into weird species, but the problem is that you cannot transform them again. So transforming someone into a Vox/Diona/Slime/Plasmaman is irreversible, you cannot Transformation Sting them again because of NO_BLOOD (Diona, Plasmaman), NO_SCAN (Slime, Vox). You can't use any stings on those species, and I assume you also can't absorb them (haven't tested absorb).

So the transformations themselves seem fine to me, it shouldn't be too hard to allow those species to be stung and absorbed.

However turning people into Plasmamen is a bit OP, Vox is fine because the Toxin Damage isn't really that bad, you can easily get to Medbay or Atmos or Science for a tank, but Plasmamen have their own suits which aren't obtainable outside of spawning as one, so the suits would need to be added to EVA or Cargo, and the Oxy and Fire damage you take will crit you in 10 seconds.

So perhaps make Transformation Sting check for Plasmaman DNA and not carry out the sting, or just not allow Plasmamen to be changelings, absorbed or stung just like IPCs.

I would like Vox, Slime People, and Dionae to all be able to spawn as changelings, be stung and be absorbed.

**Comment by LittleBigKid2000** [comment]
Not sure how I would go about making Plasmamen unable to be absorbed without snowflaking it or making a new flag just for this.

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -63,6 +63,7 @@
 #define TAIL_WAGGING    	1024
 #define NO_EYES				2048
 #define HAS_FUR				4096
+#define NO_ABSORB			8192
```
Comment (uraniummeltdown) [inline code/__DEFINES/flags.dm]: Not sure about NO_ABSORB here, I think it should be under species flags, doesn't really fit in with the body ones

File: code/game/gamemodes/changeling/changeling.dm [hunk 2]
```diff
@@ -327,4 +323,4 @@ var/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","Epsilon"
 	return 1
 
 /proc/check_species_absorb(datum/species/S)
-  return !((S.flags & NO_DNA) || (S.flags & NO_SCAN) || (S.flags & NO_BLOOD))
+  return !((S.flags & NO_ABSORB))
```
Comment (uraniummeltdown) [inline code/game/gamemodes/changeling/changeling.dm]: I think line 326 might need an isSynthetic or NO_DNA check

**Comment by uraniummeltdown** [comment]
The reason I view this PR as a fix is that it doesn't really affect balance, species that couldn't be absorbed by a ling could still be spaced, beheaded, etc. Right now any victims of Transformation Sting who get Voxed or Dionad will be stuck as that race forever unless they get a brain transplant, it makes no sense that they can be transformed one-way. So if the crew are being turned into these races the number of people able to be absorbed or whatever decreases gradually, This PR fixes those issues, and previously Slime People were excluded from ling because their blood would mess up transformations and result in instant death however this is no longer the case and they work fine.

**Comment by Fox-McCloud** [comment]
After discussing it with @TheDZD this is a change that just can't be merged in as it's balance altering. It'll have to wait post-freeze.

Brrrr

**Pull request closed without merging**